pkg/ddevapp/ddevapp_test.go

```diff
@@ -561,6 +562,58 @@ func TestDdevStartUnmanagedSettings(t *testing.T) {
 
 }
 
+// TestDdevNoFileMount tests running without the app file mount.
+func TestDdevNoFileMount(t *testing.T) {
+	assert := asrt.New(t)
+	app := &ddevapp.DdevApp{}
+
+	// Make sure this leaves us in the original test directory
+	testDir, _ := os.Getwd()
+	//nolint: errcheck
+	defer os.Chdir(testDir)
+
+	// Use Drupal8 only, mostly for the composer example
+	site := FullTestSites[1]
+	// If running this with GOTEST_SHORT we have to create the directory, tarball etc.
+	if site.Dir == "" || !fileutil.FileExists(site.Dir) {
+		app := &ddevapp.DdevApp{Name: site.Name}
+		_ = app.Stop(true, false)
+		_ = globalconfig.RemoveProjectInfo(site.Name)
+
+		err := site.Prepare()
+		require.NoError(t, err)
+		// nolint: errcheck
+		defer os.RemoveAll(site.Dir)
+	}
+	switchDir := site.Chdir()
+	defer switchDir()
+
+	runTime := util.TimeTrack(time.Now(), fmt.Sprintf("%s DdevStart", site.Name))
+	defer runTime()
+
+	err := app.Init(site.Dir)
+	assert.NoError(err)
+
+	app.OmitAppWebMount = true
+	err = app.WriteConfig()
+	assert.NoError(err)
+
+	err = app.Start()
+	assert.NoError(err)
+	//nolint: errcheck
+	defer app.Stop(true, false)
+
+	opts := &ddevapp.ExecOpts{
+		Service: "web",
+		Cmd:     "ls",
+		Dir:     "/var/www/html",
+	}
```

Comment:
This doesn't check to see if there's a mount there.  I think you'd be better to `df /var/www/html` and check the "Mounted on" to see if it's /var/www/html.  So for example, `df /var/www/html | awk '$1 != "Filesystem" {print $6}' - maybe there's a better way.

Reply:
I thought about that. The files that are there definitely wouldn't be if there _is_ a mount. Happy to test via `df` or something though. It's on my list.